key_value_service_user_defined_functions.md

```diff
@@ -171,118 +171,172 @@ For details on the broader trust model, see the [trust model explainer](https://
 
 ## APIs
 
-
 ### User-defined function API (v1)
 
+#### Generic API
 
-#### Input
+The base UDF API is generic and can be customized with  API overlays for different use cases (see [API doc](https://github.com/privacysandbox/fledge-key-value-service/blob/release-0.14/docs/APIs.md#api-all-in-one-user-guide)).
 
-The input to the UDF is a JSON object containing the
-request’s context and a partition’s key group objects.
-The format of the key groups are the same as in the v2
-Query API.
+##### Input
 
+Each UDF is expected to have the following signature:
```

Comment:
not exactly. it's more accurate to say, The generic signature is the following:
...
But if the number of arguments is known, such as in certain use cases, the UDF signature can also directly use them. For example, if it is known that the UDF will take 3 arguments:
String myHandlerName(UDFExecutionMetadata arg1, [String] some_input, [Object] some_other_input)

Reply:
Done.